Library/Homebrew/cmd/commands.rb

```diff
@@ -27,18 +27,30 @@ def commands
   end
 
   def internal_commands
-    (HOMEBREW_LIBRARY_PATH/"cmd").children.select(&:file?).map { |f| f.basename(".rb").to_s }
+    find_internal_commands HOMEBREW_LIBRARY_PATH/"cmd"
   end
 
   def internal_development_commands
-    (HOMEBREW_LIBRARY_PATH/"dev-cmd").children.select(&:file?).map { |f| f.basename(".rb").to_s }
+    find_internal_commands HOMEBREW_LIBRARY_PATH/"dev-cmd"
```

Comment:
I'm in favor of supporting them in `bin/brew` so there's no confusing difference between Ruby and Shell command support. Thus, this could stay as-is.

I wouldn't mind creating a PR for `bin/brew` to add `dev-cmd` support there, if I'm not the only one who thinks this is a good idea. (I should probably wait for #48197 to be merged first.)